subprojects/language-java/src/integTest/groovy/org/gradle/javadoc/JavadocIntegrationTest.groovy

```diff
@@ -61,6 +62,31 @@ class JavadocIntegrationTest extends AbstractIntegrationSpec {
         file("build/docs/javadoc/Foo.html").text.contains("""Hey Joe!""")
     }
 
+    @Issue("gradle/gradle#1090")
+    @Unroll
+    def "escape single quote in options enclosed with #quote"() {
+        given:
+        buildFile << """
+            apply plugin: "java"
+            javadoc.options.header = $optionIn
+        """
+
+        file("src/main/java/Foo.java") << "public class Foo {}"
+
+        when:
+        succeeds 'javadoc'
+
+        then:
+        file("build/docs/javadoc/Foo.html").text.contains(optionOut)
+
+        where:
+        quote    | optionIn                                   | optionOut
```

Comment:
I am a bit confused by all the escaping and some of the naming here. I suggest the following:

- We will not use triple, double quotes for the Strings as we'd only need it for multi-line String. That's not the case here. Let's use `\"` instead. The first test would look as such: `"'"      | "\"'single quoted'\""                          | "'single quoted'"`
- What exactly do you mean by double or triple quoted? Do you mean the use of 2 vs. 3 quotes within a String? If that's not the case, can you please clean up the test case so it becomes apparent?
- When exactly would you use `/'` to indicate a single quote? Can you clarify?